### ISSUE TITLE
Feature request: separate captions for SDXL text encoders

### ISSUE THREAD

**Issue by RockTheCosmos** [bmaltais/kohya_ss#1431]
From what I have read about the 2 text encoders that SDXL uses, the G CLIP encoder is better at understanding natural human language/full sentences, whereas the L CLIP encoder isn't as advanced so it's better to use more simplified language like tags separated with commas. From this, I deduce that it would be better to feed a WD14 caption file to the L CLIP encoder and a BLIP caption file to the G CLIP encoder.

Looking through the code, it looks like kohya_ss is currently just taking the caption from a single file and throwing that caption to both text encoders. I think it would be more effective to make it so the program can handle 2 caption files for each image, one intended for one text encoder and one intended for the other. Or you could just use a single caption file and provide a way to demarcate which part of the caption goes to one text encoder vs. the other.

You recommended in the readme that we shouldn't even train the text encoders due to unpredictable results based on the two text encoders. I think what I suggested would go a long way toward making the results more predictable.

Basically, I would just like a way to feed the program different captions for each image that are intended for one text encoder vs. the other.

**Comment by bmaltais** [comment]
Gread idea. I suggest opening an issue on @kohya-ss repo to suggest it. You can add it to the pile here: https://github.com/kohya-ss/sd-scripts/issues

I only put a GUI in front of kohya's code so such a change really need to be requested on his repo.
